### PR TITLE
chore(broker): adds snapshot metrics

### LINKED ISSUE #3447: Add RocksDB snapshot metrics
**Is your feature request related to a problem? Please describe.**

We currently have no metrics on RocksDB snapshots, which could be crucial in determining the behaviour of our brokers (i.e. replication, impact of large states, compaction, etc.)

**Describe the solution you'd like**

Adding metrics for the following:

1. Snapshot count
1. Snapshot sizes
1. Snapshotting runtime

Optionally, also the current runtime size would be interesting.

**Describe alternatives you've considered**

**Additional context**

To be done after we merge our log iteration.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Adds the following snapshot-related metrics per partition:

- Snapshot count (gauge)
- Snapshot size (gauge)
- Snapshot file size (histogram) for a distribution of file sizes
- Snapshot operation duration (gauge) (i.e. `RocksDB#checkpoint`)
- Ongoing snapshot replication (gauge), from the POV of the consumer
- Snapshot replication duration (gauge), measured on completion by the consumer

## Related issues

closes #3447 
blocked by [zeebe-io/atomix#141](https://github.com/zeebe-io/atomix/pull/141)

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by npepinpe** [comment]
How the new metrics look in Grafana:

Screenshot from 2020-01-15 14-53-53
Screenshot from 2020-01-15 14-54-07

**Review by deepthidevaki — APPROVED** [review]
:+1: can be merged once the PR on atomix is merged.

**Pull request merged**
